docs/content/preview/contribute/docs/syntax-diagrams.md

```diff
@@ -9,376 +9,134 @@ menu:
-  </div>
-</div>
+```ebnf
+(* Supporting rules *)
 ```
 
-You take responsibility for spelling this
+### Grammar diagrams file
 
-```output
-../../../syntax_resources/dir_1/dir_2/dir_3/<rule set X>.grammar.md
-```
+The grammar [diagrams file](../../../api/ysql/syntax_resources/grammar_diagrams/#abort) contains every [_syntax diagram_](#syntax-diagram) that is generated from all of the [_syntax rules_](#syntax-rule) that are found in the [_diagram definition file_](#diagram-definition-file).
 
-and this:
+This file is automatically generated.
 
-```output
-../../../syntax_resources/dir_1/dir_2/dir_3/<rule set X>.diagram.md
-```
+## Add or modify syntax diagrams
 
-properly. The rest is entirely boilerplate. You musn't touch it. The construct `../../../` must climb up the proper number of levels (three in this example) to reach the parent [_ysql directory_](#ysql-directory) from `wants-to-include.md`.
+[Syntax diagrams](#syntax-diagram) are generated dynamically. The diagram/grammar pair are automatically populated based on the [syntax rules](#syntax-rule).
 
-Notice these two matching tag pairs:
+There are typically two workflows:
 
-- _\<a href="#grammar" ..._ with _\<div id="grammar" ..._
-- _\<a href="#diagram" ..._ with _\<div id="diagram" ..._
+1. Add or modify syntax rules.
 
-Sometimes, you'll want to include more than one [_free-standing generated grammar-diagram pair_](#free-standing-generated-grammar-diagram-pair) in a particular [_humanly typed documentation source_](#humanly-typed-documentation-source) `.md` file. In such a case, you must use differently-spelled matching tag pairs for each successive diagram. The convention is to use these pairs for the second grammar-diagram pair:
+    You do this by editing the [_diagram definition file_](#diagram-definition-file).
 
-- _\<a href="#grammar-2" ..._ with _\<div id="grammar-2" ..._
-- _\<a href="#diagram-2" ..._ with _\<div id="diagram-2" ..._
+1. Add or modify the syntax diagrams to be displayed on a page.
 
-and to use these for the third diagram pair:
+    You do this by adding an `ebnf` block to the page.
 
-- _\<a href="#grammar-3" ..._ with _\<div id="grammar-3" ..._
-- _\<a href="#diagram-3" ..._ with _\<div id="diagram-3" ..._
+### Add and modify syntax rules
 
-and so on. If you do this, then _all_ diagram-pairs on the rendered page will switch in lock-step between the _"Grammar"_ tab and the _"Diagram"_ tab when you click the tab to make your choice using _any one_ of the pairs. Failure to follow this rule will mean that the syntax tabs don't display correctly.
+To add a new syntax rule or modify an existing one, edit the [_diagram definition file_](#diagram-definition-file).
 
-### Diagram generator
+{{<note title="Note">}}
+If you are developing on a local machine, you need to restart Hugo after modifying the [_diagram definition file_](#diagram-definition-file) for the changes to be reflected on the local preview.
```

Comment:
This might be a very bad experience if you don't know what will be valid syntax. Hugo takes ~80s to restart for me (compiles all docs for all versions).

Reply:
Added some extra text here to warn

Reply:
@ddorian , we have added a new command `npm run fast` to just build the preview and stable versions for dev mode. I'll make a note of it here.